### ISSUE TITLE
"Elegant" way to check if string::find() failed?

### ISSUE THREAD

**Issue by totalgee** [ChaiScript/ChaiScript#347]
* Compiler Used: Visual Studio 2015
 * Operating System: Windows 10
 * Architecture (ARM/x86/32bit/64bit/etc): Intel i7

I expected that maybe `npos` might be exposed by ChaiScript, since it exposes `string::find()`.  None of the string unit tests show an example of a failed match (I often look there to see tips on how to do things in ChaiScript).  Since the string search methods return `size_t`, I guess the "not so elegant" way to test for failed string find right now would be:

```c++
auto findResult = "Hello".find("there");
if (findResult == size_t(-1)) { // in C++ we'd use std::npos here
    // didn't find "there"...
}
```
(A test to see if `findResult < 0` doesn't work, because `size_t` is unsigned.)

Not a big deal, obviously, I guess I'll just expose the constant "npos" in my own application, but it might be an idea to add it to ChaiScript's STL bootstrap bindings.

**Comment by lefticus** [comment]
Agreed - npos should probably be exposed. The problem I guess is the lack of namespacing support. I still need to get back to that.

**Comment by olikraus** [comment]
Came accross the same problem. The interessting thing is, that
```
if ( findResult == -1 )
```
works, but (as mentioned above)
```
if ( findResult < 0 )
```
will fail.

**Comment by totalgee** [comment]
@olikraus For the second comparison, it fails because `string::find()` returns an unsigned integer type (specifically of type `size_t`) -- so it can therefore never be less than zero.

In your first comparison, the literal `-1` gets converted to the largest possible representable value of `size_t` (which is likely equivalent to `uint64_t` or `uint32_t`, depending on the platform you've compiled for).  `std::string::npos` is also defined in the `<string>` header as -1, so it actually holds that same "largest possible value" -- so the comparison works.

My point was it would be nice to have a ChaiScript constant defined for `npos` (or maybe `string.npos` in ChaiScript).  In my own code, I did just that. But using -1 is fine, too.

**Comment by olikraus** [comment]
So you added something like
```
def npos(string s)
{
  return size_t(-1);
}
```

Ok, indeed the following will print "true":
```
./chai -c 'def npos(string s) { return size_t(-1); } var s="xyz"; print(s.find("a") == s.npos);'
```

**Comment by totalgee** [comment]
More or less, although I actually decided to expose something like this in C++ instead:

```c++
    chai.add_global_const(chaiscript::const_var(string::npos), "string_npos");
```
